tests/1_engine/rex_engine_unit_test/src/test_path.cpp

```diff
@@ -0,0 +1,525 @@
+    CHECK(rex::path::is_relative(".") == false);  // Current directory
+    CHECK(rex::path::is_relative("..") == false); // Parent directory
+}
+
+TEST_CASE("Is File")
+{
+  CHECK(rex::path::is_file("this_is_a_file.txt"));
+  CHECK(rex::path::is_file("this_is_a_directory") == false);
+  CHECK(rex::path::is_file(rex::path::random_filename()) == false);
+  CHECK(rex::path::is_file(rex::path::random_dir()) == false);
+}
+
+TEST_CASE("Is Directory")
+{
+  CHECK(rex::path::is_dir("this_is_a_file.txt") == false);
+  CHECK(rex::path::is_dir("this_is_a_directory"));
+  CHECK(rex::path::is_dir(rex::path::random_filename()) == false);
+  CHECK(rex::path::is_dir(rex::path::random_dir()) == false);
+}
+
+TEST_CASE("Is Junction")
+{
+  CHECK(rex::path::is_junction("original_folder") == false);
+  CHECK(rex::path::is_junction("original_file.txt") == false);
+  CHECK(rex::path::is_junction("junction_folder") == true);
+}
+
+TEST_CASE("Is Link")
+{
+  CHECK(rex::path::is_link("original_folder") == false);
+  CHECK(rex::path::is_link("original_file.txt") == false);
+  CHECK(rex::path::is_link("junction_folder") == false);
+}
+
+TEST_CASE("Is Same File")
+{
+  [[maybe_unused]] const bool b = rex::path::exists("same_file.txt");
+
+  CHECK(rex::path::same_path("same_file.txt", "same_file.txt") == true);
+  CHECK(rex::path::same_path("sub_folder_1/../same_file.txt", "same_file.txt") == true);
+  CHECK(rex::path::same_path("same_file.txt", rex::path::random_filename()) == false);
+  CHECK(rex::path::same_path("same_file.txt", rex::path::random_dir()) == false);
+
+
+  CHECK(rex::path::same_path("sub_folder_1", "sub_folder_1") == true);
+  CHECK(rex::path::same_path("sub_folder_1/../sub_folder_1", "sub_folder_1") == true);
+  CHECK(rex::path::same_path("same_file.txt", rex::path::random_filename()) == false);
+  CHECK(rex::path::same_path("same_file.txt", rex::path::random_dir()) == false);
+
+  CHECK(rex::path::same_path(rex::path::random_filename(), rex::path::random_filename()) == false);
+  CHECK(rex::path::same_path(rex::path::random_filename(), rex::path::random_dir()) == false);
+  CHECK(rex::path::same_path(rex::path::random_dir(), rex::path::random_dir()) == false);
+}
+
+TEST_CASE("Split")
+{
+    auto result = rex::path::split("/path/to/file.txt");
+    CHECK(result.head == "/path/to");
+    CHECK(result.tail == "file.txt");
+
+    result = rex::path::split("C:\\Windows\\System32\\file.dll");
+    CHECK(result.head == "C:\\Windows\\System32");
+    CHECK(result.tail == "file.dll");
+
+    result = rex::path::split("./relative/path");
+    CHECK(result.head == "./relative");
+    CHECK(result.tail == "path");
+
+    result = rex::path::split("");
+    CHECK(result.head == "");
+    CHECK(result.tail == "");
+
+    result = rex::path::split(".");
+    CHECK(result.head == "");
+    CHECK(result.tail == ".");
+
+    result = rex::path::split("..");
+    CHECK(result.head == "");
+    CHECK(result.tail == "..");
+}
+
+TEST_CASE("Split Drive")
```

Comment:
I would call this " scheme " or " volume " cuz it will not always be a drive. 
what if it's a URL: http://